tstest/integration/integration_test.go

```diff
@@ -405,12 +557,27 @@ func (d *Daemon) MustCleanShutdown(t testing.TB) {
 
 // StartDaemon starts the node's tailscaled, failing if it fails to
 // start.
-func (n *testNode) StartDaemon(t testing.TB) *Daemon {
-	cmd := exec.Command(n.env.Binaries.Daemon,
-		"--tun=userspace-networking",
-		"--state="+n.stateFile,
-		"--socket="+n.sockFile,
-	)
+// Additional flags to tailscaled can be passed through specialArgs
+func (n *testNode) StartDaemon(t testing.TB, specialArgs ...string) *Daemon {
+	var cmd *exec.Cmd
+	// Support additional flags in starting up tailscaled
+	if len(specialArgs) > 0 {
```

Comment:
no need for two duplicated blocks here. It's safe to do the `flags = append(flags, specialArgs...)` if specialArgs is length 0 (including nil)